realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/internal/async/RealmResultTaskImplTest.kt

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2020 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.internal.async
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import io.realm.mongodb.App
+import io.realm.mongodb.AppException
+import io.realm.mongodb.RealmResultTask
+import io.realm.rule.BlockingLooperThread
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.util.concurrent.ThreadPoolExecutor
+import kotlin.test.*
+
+private const val OUTPUT = 42
+private const val EXCEPTION_REASON = "BOOM"
+
+@RunWith(AndroidJUnit4::class)
+class RealmResultTaskImplTest {
+
+    private val looperThread = BlockingLooperThread()
+    private val service: ThreadPoolExecutor = App.NETWORK_POOL_EXECUTOR
+
+    @Test
+    fun constructor_throwsOnNullArgs() {
+        assertFailsWith<IllegalArgumentException> {
+            RealmResultTaskImpl<String>(null, object : RealmResultTaskImpl.Executor<String>() {
```

Comment:
It is a bit concerning that `null` is allowed outright here. Normally we attempt to have default nonnullable using package annotations like this: https://github.com/realm/realm-java/blob/master/realm/realm-library/src/main/java/io/realm/package-info.java

With that enabled you can use `TestHelper.getNull()` here instead, since adding that package level annotation will otherwise make Kotlin treat it as nonnull pr default.

Reply:
That's right, I keep forgetting about that annotation.